cmd/query/app/apiv3/otlp_translator.go

```diff
@@ -45,3 +46,26 @@ func modelToOTLP(spans []*model.Span) ([]*tracev1.ResourceSpans, error) {
 	}
 	return chunk.ResourceSpans, nil
 }
+
+func OTLP2model(OTLPSpans []byte) ([]model.Trace, error) {
+	ptraceUnmarshaler := ptrace.JSONUnmarshaler{}
+	otlpTraces, err := ptraceUnmarshaler.UnmarshalTraces(OTLPSpans)
+	if err != nil {
+		fmt.Println(err)
+		return nil, fmt.Errorf("cannot unmarshal OTLP : %w", err)
+	}
+	jaegerBatches, err := model2otel.ProtoFromTraces(otlpTraces)
+	if err != nil {
+		fmt.Println(err)
+		return nil, fmt.Errorf("cannot transform OTLP to Jaeger: %w", err)
+	}
+	var jaegerTraces []model.Trace
+	for _, v := range jaegerBatches {
```

Comment:
This may be incorrect. OTEL uses the notion of "resource" that is equivalent to Jaeger's Process. `otlpTraces` is a collection of `resourceSpans` where each entry encapsulates a single resource and a bunch of spans - this is equivalent to Jaeger's Batch. We need to check how `model2otel.ProtoFromTraces` works, but it's quite possible that is just maps a single `resourceSpans` entry into a single Batch, since that's the equivalent granularity. Which means at this point even you have multiple batches you may still have one trace, and vice versa, a single batch may contain spans from different traces. This is all because these structures are optimized for ingestion, not retrieval. So the correct way would be to tease apart all batches into individual spans and group-by those spans by trace-id, only then will you be sure.

Reply:
btw, please make sure you have unit tests for all new functionality.

Reply:
Would it be fine to do this process manually? by maybe pushing these spans in a map of (trace_id, span) kv pair and later combining these spans into a single trace?

Reply:
> btw, please make sure you have unit tests for all new functionality.

Im not sure how to test the http endpoint thats supposed to have OTLP type traces